packages/header-change-detection/lib/header-change-detection.ts

```diff
@@ -0,0 +1,114 @@
+import { DockerImage, Duration } from "aws-cdk-lib";
+import { AttributeType, BillingMode, Table } from "aws-cdk-lib/aws-dynamodb";
+import { Rule, RuleProps, Schedule } from "aws-cdk-lib/aws-events";
+import { LambdaFunction, SnsTopic } from "aws-cdk-lib/aws-events-targets";
+import { Architecture, Code, Function, Runtime } from "aws-cdk-lib/aws-lambda";
+import { Construct } from "constructs";
+import { join } from "path";
+import { Esbuild } from "@aligent/cdk-esbuild";
+
+export interface HeaderChangeDetectionProps {
+  /**
+   * Optional list of additional headers to monitor
+   * 
+   * @default []
+   */
+  additionalHeaders?: string[];
+
+  /**
+   * Optionally disable all the default headers
+   * 
+   * @default false
+   */
+  disableDefaults?: boolean;
+
+  /**
+   * SNS Topic to send change detection notifications to
+   */
+  snsTopic: SnsTopic
+
+  /**
+   * The schedule for performing the header check
+   * 
+   * @default Schedule.rate(Duration.hours(1))
+   */
+  schedule?: Schedule
+
+  /**
+   * Optionally pass any rule properties 
+   */
+  ruleProps?: Partial<RuleProps>
+}
+
+const command = [
+  "sh",
+  "-c",
+  'echo "Docker build not supported. Please install esbuild."',
+];
+
+const defaultHeaders = [
+  'content-security-policy',
+  'content-security-policy-report-only',
+  'reporting-endpoints',
+  'strict-transport-security',
+  'x-frame-options',
+  'x-content-type-options',
+  'cross-origin-opener-policy',
+  'cross-origin-embedder-policy',
+  'cross-origin-resource-policy',
+  'referrer-policy',
+  'permission-policy',
+  'cache-control',
+  'set-cookie',
+]
+
+export class HeaderChangeDetection extends Construct {
+
+  constructor(scope: Construct, id: string, props: HeaderChangeDetectionProps) {
+    super(scope, id);
+
+    const headers = props.disableDefaults ? [] : defaultHeaders;
+
+    headers.push(...props.additionalHeaders?.map(header => header.toLowerCase()) || []);
+
+    const table = new Table(this, 'Table', {
+      partitionKey: {
+        name: 'Url',
+        type: AttributeType.STRING
+      },
+      billingMode: BillingMode.PAY_PER_REQUEST
+    })
+
+    const schedule = new Rule(this, 'EventRule', {
+      schedule: props.schedule || Schedule.rate(Duration.hours(1)),
+      ...props.ruleProps
+    });
+
+    const lambda = new Function(this, 'HeaderCheck', {
+      architecture: Architecture.X86_64,
+      runtime: Runtime.NODEJS_20_X,
+      handler: "header-check.handler",
+      code: Code.fromAsset(join(__dirname, "lambda"), {
+        bundling: {
+          command,
+          image: DockerImage.fromRegistry("busybox"),
+          local: new Esbuild({
+            entryPoints: [join(__dirname, "lambda/header-check.ts")],
+          })
+        }
+      }),
+      environment: {
+        "URLS": "https://production.pwa.aligent.com.au/,https://production.pwa.aligent.com.au/checkout",
```

Comment:
This should be coming through as a prop right?

Reply:
Oops, yep!